data/lib/core/creature.lua

```diff
@@ -50,3 +50,48 @@ end
 function Creature.isTile(self)
 	return false
 end
+
+function Creature:setMonsterOutfit(monster, time)
+	if not self then
+		return false
+	end
+
+	local monsterType = MonsterType(monster)
+	if not monsterType then
+		return false
+	end
+
+	if self:getPlayer() and not getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) then
+		if not monsterType:isIllusionable() then
```

Comment:
Been a while since I wrote this but my guess is that it would be to long, this pr has been open for ages so I doubt it will even be merged during 2017 ;)

Once mark has time to check my prs I can update them, no reason to waste time when ill have to rebase in x months, but thanks for the tip and now I have a reminder of it :p